packages/lodestar/src/sync/backfill/backfill.ts

```diff
@@ -821,6 +825,21 @@ export class BackfillSync extends (EventEmitter as {new (): BackfillSyncEmitter}
     }
     if (error) throw new BackfillSyncError({code: error});
   }
+
+  private async archiveBlocksTreeBacked(blocks: TreeBacked<allForks.SignedBeaconBlock>[]): Promise<void> {
+    const blockEntries: BlockArchiveBatchPutBinaryItem[] = blocks.map((signedBlock) => {
+      const block = signedBlock.message as TreeBacked<allForks.BeaconBlock>;
+      return {
+        key: block.slot,
+        value: signedBlock.serialize(),
```

Comment:
what I mean in the issue is that we return both the binary and the signed block in `beaconBlocksByRange` and `beaconBlocksByRoot`. So instead of returning `Promise<allForks.SignedBeaconBlock[]>` in network req/resp api, return something like

```typescript
type CachedBytes<T> = {bytes: Uint8Array} & T;

async beaconBlocksByRange(
    peerId: PeerId,
    request: phase0.BeaconBlocksByRangeRequest
  ): Promise<CachedBytes<allForks.SignedBeaconBlock>[]>
```

and consume that cached `bytes` without having to call `serialize()` again in BackFill sync and similar places

@dapplion do we have this type in ssz v2?

Reply:
yes, I was wondering about the same :slightly_smiling_face:
This format should currently save on hashTreeRoot calcs which would amount for 9% of hashTreeRoot in verifiBlockSequence (https://github.com/ChainSafe/lodestar/issues/3657#issuecomment-1019160634), and may be some 2-3% on the serialize. 

CachedBytes<allForks.SignedBeaconBlock> would be awesome :heart: awaiting @dapplion 's input regarding ssz v2.

Reply:
Right now, ssz v2 only (optionally) caches the hashTreeRoot (of non-tree-backed values); doesn't cache the serialized bytes.

Reply:
https://github.com/ChainSafe/ssz/pull/223/files#diff-bcfddac4baebdcc3adc704a754ecd03d1b5f8c78ba7234f3ac088e4042337e06R32-R38

Reply:
added caching capability as that was discussed to be out of scope for ssz v2 :+1: